metadata/metadata.go

```diff
@@ -145,8 +153,12 @@ func NewOutgoingContext(ctx context.Context, md MD) context.Context {
 }
 
 // AppendToOutgoingContext returns a new context with the provided kv merged
-// with any existing metadata in the context. Please refer to the
-// documentation of Pairs for a description of kv.
+// with any existing metadata in the context. Please refer to the documentation
+// of Pairs for a description of kv.
+//
+// Unlike Pairs, the keys are not turned into lowercase immediately. Users of
+// FromOutgoingContextRaw and FromOutgoingContext need to handle them
```

Comment:
Don't document FromOutgoingContextRaw please; we don't want anyone using it.

Reply:
Deleted